res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -38,97 +38,124 @@ const KeyboardColors = [
     LedColors.white,
 ];
 
+
+/**
+ * This helper function is used to ensure backward compatibility between 2.4 and later versions
+ * @param {string} key the setting key as per defined in the manifest
+ * @param {any} defaultVal the value to use if not controller setting is available
+ * @returns {any} the value to use for this setting
+ */
+var getSetting = (key, defaultVal) => {
+    if (engine.getSetting) {
+        const value = engine.getSetting(key);
+        if (typeof value !== typeof defaultVal) {
+            console.warn(`expected a value of type ${typeof defaultVal} for "${key}", got a ${typeof value} ("${value}"). Using default value.`);
+            return defaultVal;
+        }
+        return value;
+    }
+    return engine.getSetting ? engine.getSetting(key) : defaultVal;
+};
+
 /*
  * USER CONFIGURABLE SETTINGS
- * Adjust these to your liking
+ * Adjust these to your liking or change settings in the preferences (Mixxx >= 2.5)
+ * Make sure to edit the second argument, or second value after the coma
  */
 
 const DeckColors = [
-    LedColors.red,
-    LedColors.blue,
-    LedColors.yellow,
-    LedColors.purple,
+    LedColors[getSetting("deckA", "red")],
+    LedColors[getSetting("deckB", "blue")],
+    LedColors[getSetting("deckC", "yellow")],
+    LedColors[getSetting("deckD", "purple")],
 ];
 
 const LibrarySortableColumns = [
-    script.LIBRARY_COLUMNS.ARTIST,
-    script.LIBRARY_COLUMNS.TITLE,
-    script.LIBRARY_COLUMNS.BPM,
-    script.LIBRARY_COLUMNS.KEY,
-    script.LIBRARY_COLUMNS.DATETIME_ADDED,
-];
+    getSetting("librarySortableColumns1Value", script.LIBRARY_COLUMNS.ARTIST.toString()),
+    getSetting("librarySortableColumns2Value", script.LIBRARY_COLUMNS.TITLE.toString()),
+    getSetting("librarySortableColumns3Value", script.LIBRARY_COLUMNS.BPM.toString()),
+    getSetting("librarySortableColumns4Value", script.LIBRARY_COLUMNS.KEY.toString()),
+    getSetting("librarySortableColumns5Value", script.LIBRARY_COLUMNS.DATETIME_ADDED.toString()),
+    getSetting("librarySortableColumns6Value", "0"),
+].map(c => parseInt(c)).filter(c => c); // Filter '0' column, equivalent to '---' value in the UI or disabled
```

Comment:
You would have expected `.map(parseInt)` to work, write?


Probably one I should add [here](https://github.com/denysdovhan/wtfjs?tab=readme-ov-file#parseint-is-a-bad-guy)

Reply:
yay  javascript lol sob